pytensor/tensor/math.py

```diff
@@ -602,37 +602,248 @@ def isneginf(x):
 
 @scalar_elemwise
 def lt(a, b):
-    """a < b"""
+    """a < b
+
+    Computes element-wise less than comparison between two tensors.
+
+    Parameters
+    ----------
+    a : TensorLike
+        First input tensor
+    b : TensorLike
+        Second input tensor
+
+    Returns
+    -------
+    tensor
+        Output tensor of type bool, with 1 (True) where a < b,
+        and 0 (False) elsewhere.
+
+    Examples
+    --------
+    >>> import pytensor
+    >>> import pytensor.tensor as pt
+    >>> x = pt.vector("x")
+    >>> y = pt.vector("y")
+    >>> f = pytensor.function([x, y], pt.lt(x, y))
+    >>> f([1, 2, 3], [2, 2, 2])
+    array([ True, False, False])
+
+    Notes
+    -----
+    Due to Python rules, it is not possible to correctly overload the less than symbol `<` for hashable objects,
+    so `lt` must always be used to compute the Elemwise less than comparison of TensorVariables (which are hashable).
```

Comment:
This is only the case for eq and neq, <,<=, >, >= work fine with PyTensor variables